src/main/java/io/zksync/wallet/DefaultZkASyncWallet.java

```diff
@@ -36,6 +37,8 @@
 
     private final TransactionBuildHelper helper;
 
+    private Tokens tokens;
```

Comment:
Maybe we should cache the tokens inside the provider?

This is how it is done in the JS library and it seems to me more universal